### DISCUSSION TITLE
Requesting assistance on how to use the tag alias > actions tab > script function to generate alerts.

### DISCUSSION THREAD

**Discussion by Justinwb86** [DSheirer/sdrtrunk#1891] [Q&A]
How would you go about generating a message or potentially send a recording to an email, text, or preferably a push notification when a certain Talkgroup or radio ID is detected?  I have seen some talk about it generally but not how the scripting is used or any examples.  The function I am referring to is under tag alias > actions tab > script in the playlist editor.  Any help would be appreciated!

**Comment by DSheirer** [comment]
Create a script file that performs some kind of action and assign that script file to the alias action.

This feature is somewhat limited because you currently can't pass any of the details of the event or the participating radios/talkgroups as parameters to the script.  In the future I'd like to be able to pass parameters to the script so that you could achieve some of the things you're looking to do.

**Reply by hjellinek** [reply]
I had the same question.  I've followed these instructions above and created a shell script (I'm running on macOS) that writes a status message to a temp file:

```
#!/bin/bash
(echo Script running.; echo Args: "$@"; echo Done.) >> /tmp/log.txt
```

Because I have no P-25 systems nearby, I set up the local PD's channels and the local National Weather Service channel (all NBFM) to invoke the action when active.  Since NWS is always active I figured the script would run as soon as I clicked Play on the channel.

Thee script runs as expected when I click the `Test` button, but it never runs in response to a call on a channel.

Is my configuration wrong?

I'm running SDRTrunk I built myself from `master` as of today, June 26 2025.

I've attached screen captures showing the National Weather Service configuration.  The police configuration is similar.

![20250626_134519_screen_capture](https://github.com/user-attachments/assets/022ac49a-d018-47bc-a4de-fc2c20ba9d8a)

![20250626_134731_screen_capture](https://github.com/user-attachments/assets/62c3653a-37a6-4fdf-8365-f7b487928ef8)

![20250626_134902_screen_capture](https://github.com/user-attachments/assets/c64d8fa5-ab3d-4f82-b075-5e00f4522680)

Thanks.
